recipes-hipos/hipos-time/files/set-time.sh

```diff
@@ -1,19 +1,61 @@
 #!/bin/bash
 
-MACHINE_TYPE="$(/usr/sbin/hip-machinfo -a)"
+log() {
+	logger -t set-time.sh $1
+}
+
 SUB_TYPE="$(/usr/sbin/hip-machinfo -s)"
-VARIANT=$(. /etc/hydraip-devid; echo "${device:-recorder}")
+
+# reset error signaling
+rm /etc/drerrors/RTC.json 2> /dev/null
 
 if [ "$SUB_TYPE" != "dvmon" -a "$SUB_TYPE" != "ipcam" ]; then
-        date -u -s "$(drbcc --dev=/dev/ttydrbcc --cmd=getrtc | cut -d\  -f2,3)" ||
-        {  # in case of problems we log the commands
-                set -x
-                for try in 1 2 3 4 5; do
-                date -u -s "$(drbcc --dev=/dev/ttydrbcc --cmd=getrtc | cut -d\  -f2,3)" && break
-                done 2>&1 | logger -t $0
-                set -x
-        }
-else
-        date -s `cat /lib/systemd/systemd-timesyncd` 2> /dev/null        
+	date=$(drbcc --dev=/dev/ttydrbcc --cmd=getrtc | cut -d\  -f2,3)
+	if [ -z "$date" ]; then
+		# in case of problems we log the commands
+		set -x
+		for try in 1 2 3 4 5; do
+			date=$(drbcc --dev=/dev/ttydrbcc --cmd=getrtc | cut -d\  -f2,3)
+			if [ -n "$date" ]; then
+				break
+			fi
+		done 2>&1 | log
+		set +x
+	fi
 fi
 
+# got last shutdown time
+last=$(cat /lib/systemd/systemd-timesyncd 2> /dev/null)
+
+# $date and $last may be empty or invalid here
+if [ -n "$date" ]; then
+	# empty battery
+	if [[ "$date" < "2019" ]]; then
+		log "Warning: RTC time outdated: $date"
+		# signal error with LED
+		echo "{\"error\":\"RTC time outdated: $date\"}" > /etc/drerrors/RTC.json
```

Comment:
It isn't save to assume that directory `/etc/drerrors` still exists.

Reply:
What do you suggest? " 2> /dev/null "?

Reply:
`mkdir -p /etc/drerrors` before writing to the file.

Reply:
Ack